proposals/secrets-support.md

```diff
@@ -0,0 +1,85 @@
+# First class secrets support in devcontainers CLI
+
+## What are secrets
+Secrets are variables that hold sensitive values and needs to be handled securely at all times (API keys, passwords etc.). Users can change a secret's value at anytime, and the devcontainers CLI should support dynamically changing secrets without having to rebuild the container.
+
+## Goal
+
+Support Secrets as a first class feature in devcontainers CLI.
+
+This feature should comprise of the ability to securely pass in the secrets in CLI commands, make them available for users to consume similar to remoteEnv and containerEnv, and be processed and handled securely at all times.
+This functionality will allow consumers to be able to use secrets more predictably and securely with devcontainers CLI.
+
+## Motivation
+
+Today Codespaces pass secret variables as `remoteEnv` since there are no other mechanisms to pass them explicitly as secrets. The devcontainers CLI do not (and need not) treat `remoteEnv` as secrets. Having explicit support for secrets will help eliminating security threats such as leaking secrets, as well as in improving customer confidence.
+
+## Proposal
+
+Following are the three main functional changes to the CLI to support this feature.
```

Comment:
I know this is change is going to immediately be specific to the reference CLI implementation, but I think with slight rewording in a few key spots this could read more like a generalized specification.  Something like (IMHO):

```suggestion
A [supporting tool](https://containers.dev/supporting#tools) (ie, the [dev container CLI reference implementation](https://github.com/devcontainers/cli) should behave according to the following properties:
```